### ISSUE TITLE
Having `&` in Directory name causign problems.

### ISSUE THREAD

**Issue by butuzov** [go-task/task#1551]
- Task version: 3.35.1
- Operating system: darwin 13.6.5 (m1)
- Experiments enabled: no idea.

Running task in watch mode with next configuration
```yaml
# https://taskfile.dev

version: '3'

vars:
  GREETING: Hello, World!

tasks:
  default:
    sources:
    - ./**/*.*
    - ./*.* 
    method: timestamp
    cmds:
      - ./update_me_please
    silent: true
```

Causes it to fail with proximatly next message

```
/Path/Teka/Father&Sone/first_path.txt: 1:64: & is not a valid word
```
So having "&" in folder is a problem...

related - https://github.com/go-task/task/discussions/1069

**Comment by vergenzt** [comment]
I've encountered this same issue but with a different special character (parentheses!), and I think I've got a minimal reproduction:

```yml
# Taskfile.yml
version: '3'

vars:
  FILE_BASENAME: 'Filename with (parens) & other special chars'

tasks:
  gen:
    generates:
    - '{{.FILE_BASENAME}}.txt'
    cmds:
    - echo foo > '{{.FILE_BASENAME}}.in'

  default:
    deps: [gen]
    #sources:
    #- '{{.FILE_BASENAME}}.in'
    #generates:
    #- '{{.FILE_BASENAME}}.txt'
    cmds:
    - cp '{{.FILE_BASENAME}}.in' '{{.FILE_BASENAME}}.txt'
```

As-is, I get:
```console
$ task
task: [gen] echo foo > 'Filename with (parens) & other special chars.in'
1:51: ( is not a valid word
```

However with the `sources` and `generates` lines from `default` task removed:
<details>
<summary>Taskfile.yml (updated)</summary>

```yml
# Taskfile.yml
version: '3'

vars:
  FILE_BASENAME: 'Filename with (parens) & other special chars'

tasks:
  gen:
    generates:
    - '{{.FILE_BASENAME}}.txt'
    cmds:
    - echo foo > '{{.FILE_BASENAME}}.in'

  default:
    deps: [gen]
    #sources:
    #- '{{.FILE_BASENAME}}.in'
    #generates:
    #- '{{.FILE_BASENAME}}.txt'
    cmds:
    - cp '{{.FILE_BASENAME}}.in' '{{.FILE_BASENAME}}.txt'
```

</details>

the task seems to work fine:
```console
$ task
task: [gen] echo foo > 'Filename with (parens) & other special chars.in'
task: [default] cp 'Filename with (parens) & other special chars.in' 'Filename with (parens) & other special chars.txt'
```

---

Interestingly... this error _only_ seems to happen when _both_ `sources` and `generates` are provided values with special characters. When I commented out either one of the fields but not the other, the whole thing worked fine!

**Comment by andreynering** [comment]
Hi @butuzov and @vergenzt,

Thanks for the reproduction steps!

I managed to fix this bug on #1584.

**Comment by vergenzt** [comment]
Awesome!!! Thanks @andreynering!

When do you think #1584 will be able to be released? I have an active project that uses lots of special characters right now. 😅
